p2p/src/lib.rs

```diff
@@ -1,6 +1,6 @@
 //! The Tendermint P2P stack.
 
-#![forbid(unsafe_code)]
+#![deny(unsafe_code)]
```

Comment:
Why changing this?

Reply:
Is there any way to avoid `unsafe` code? We're trying to have no `unsafe` code in the whole tendermint-rs repo.

Reply:
The only usage of `unsafe` in the pipe implementation is:

https://github.com/informalsystems/tendermint-rs/blob/9b7fac4e4fd720e90b76a2c7a72ebfa9ac629c9b/p2p/src/secret_connection/pipe.rs#L159-L172

AFAIU it's solely to eliminate that branch during compilation, I reckon we can refactor that away.

Reply:
Why not just `.unwrap()`?